test/hotspot/jtreg/runtime/NMT/TotalMallocMmapDiffTest.java

```diff
@@ -0,0 +1,109 @@
+    private static final WhiteBox wb = WhiteBox.getWhiteBox();
+    private static final long ALLOCATE_SIZE = 250 * 1024 * 1024; // 250MB
+    private static final double FUDGE_FACTOR = 0.2;
+    private static final double UPPER_BOUND = ALLOCATE_SIZE * (1 + FUDGE_FACTOR);
+    private static final double LOWER_BOUND = ALLOCATE_SIZE * (1 - FUDGE_FACTOR);
+
+    public static void main(String[] args) throws Exception {
+
+        // Get baseline
+        OutputAnalyzer output = NMTTestUtils.startJcmdVMNativeMemory("baseline=true");
+        output.shouldContain("Baseline taken");
+
+        // Allocate some memory via malloc
+        long addr = wb.NMTMalloc(ALLOCATE_SIZE);
+
+        // Virtually reserve and commit memory
+        addr = wb.NMTReserveMemory(ALLOCATE_SIZE);
+        wb.NMTCommitMemory(addr, ALLOCATE_SIZE);
+
+        // Get NMT diff
+        output = NMTTestUtils.startJcmdVMNativeMemory("summary.diff", "scale=1");
+
+        // Verify malloc diff accounts for memory allocation with a fudge factor
+        long mallocDiff = getMallocDiff(output);
+        if (mallocDiff < LOWER_BOUND || mallocDiff > UPPER_BOUND) {
+            throw new Exception("Total malloc diff is incorrect. " +
+                    "Expected malloc diff range: [" + LOWER_BOUND + " - " + UPPER_BOUND + "]" +
+                    "Actual malloc diff: " + mallocDiff);
+        }
+
+        // Verify mmap diff accounts for reserve and commit
+        long reservedDiff = getReservedDiff(output);
+        long committedDiff = getCommittedDiff(output);
+        if (reservedDiff < LOWER_BOUND || reservedDiff > UPPER_BOUND) {
+            throw new Exception("mmap reserved diff is incorrect. " +
+                    "Expected reserved diff range: [" + LOWER_BOUND + " - " + UPPER_BOUND + "]" +
+                    "Actual reserved diff: " + reservedDiff);
+        }
+        if (committedDiff < LOWER_BOUND || committedDiff > UPPER_BOUND) {
+            throw new Exception("mmap committed diff is incorrect. " +
+                    "Expected committed diff range: [" + LOWER_BOUND + " - " + UPPER_BOUND + "]" +
+                    "Actual committed diff: " + committedDiff);
+        }
+
+    }
+
+    private static long getMallocDiff(OutputAnalyzer output) {
+        // First match should be global malloc diff
+        String malloc = output.firstMatch("malloc=\\d* \\+\\d*");
+        return Long.parseLong(malloc.substring(malloc.indexOf("+") + 1));
+    }
+
+    private static long getReservedDiff(OutputAnalyzer output) {
+        // First match should be global mmap diff
+        String reservedDiff = output.firstMatch("mmap: reserved=\\d* \\+\\d*");
+        return Long.parseLong(reservedDiff.substring(reservedDiff.indexOf("+") + 1));
+    }
+
+    private static long getCommittedDiff(OutputAnalyzer output) {
+        // First match should be global mmap diff
+        String committedDiff = output.firstMatch("mmap: reserved=\\d* \\+\\d*, committed=\\d* \\+\\d*");
```

Comment:
Can do the same here, probably

```suggestion
        String s = output.firstMatch("mmap: reserved=\\d+ \\+\\d+, committed=\\d+ \\+(\\d+)", 1);
        return Long.parseLong(s);
```